src/App.tsx

```diff
@@ -15,36 +16,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+function sortGoods(
+  goods: string[],
+  sortField: SortType,
+  reversed: boolean,
+): string[] {
+  const sortedGoods = [...goods].sort((a, b) => {
+    if (sortField === SortType.ALPHABET) {
+      return a.localeCompare(b);
+    }
+
+    if (sortField === SortType.LENGTH) {
+      return a.length - b.length;
+    }
+
+    return 0;
+  });
+
+  return reversed ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [reversed, setReversed] = useState(false);
+
+  const reset = () => {
+    setSortField(SortType.DEFAULT);
+    setReversed(false);
+  };
+
+  const visibleGoods = sortGoods(goodsFromServer, sortField, reversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.ALPHABET,
+          })}
+          onClick={() => setSortField(SortType.ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.LENGTH,
+          })}
+          onClick={() => setSortField(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
```

Comment:
```suggestion
          onClick={() => setReversed(prevIsReversed => !prevIsReversed)}
```

https://dev.to/gakii/functional-state-update-in-react-42io